### PR TITLE
[Ready] Adding Selfchanging Assemblies

### PR DESCRIPTION
**Pull request by Shdorsh**
[Changelogs]: # It's done. It's the continuation of #9561, only post-modularization and with a few slightly added functions: you can now get the name, description and color of the assembly aswell.
:cl: Shdorsh
add: redescriber and repainter circuit
tweak: renamer circuit can read assembly name
/:cl:

[why]: It was planned and many people sobbed since I had to focus on studying and this didn't get implemented.

Edit: Tested and working.

### CODE REVIEW

**Comment by Hate9** [comment]
Can't wait for this to get put in!

File: hippiestation/code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 1]
```diff
@@ -736,12 +736,78 @@
...
 	power_draw_per_use = 1
 	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
 
-obj/item/integrated_circuit/manipulation/renamer/do_work()
-	var/new_name = get_pin_data(IC_INPUT, 1)
-	if(assembly && new_name)
-		assembly.name = new_name
-		activate_pin(2)
\ No newline at end of file
+obj/item/integrated_circuit/manipulation/renamer/do_work(var/n)
```
Comment (k-k-karma) [inline hippiestation/code/modules/integrated_electronics/subtypes/manipulation.dm]:
```suggestion
/obj/item/integrated_circuit/manipulation/renamer/do_work(var/n)
```

**Comment by Shdorsh** [comment]
K, I fixed the missing /

**Comment by Shdorsh** [comment]
This has been done and fixed for over a week, can we please get a general consensus on it?

**Comment by Carbonhell** [comment]
Hell yeah +1

**Review by Carbonhell — APPROVED** [review]

**Comment by Maroonenahmoron** [comment]
This looks tricky and complicated. Time to give it a +1 and see what happens. +1

**Comment by TheMeatGod** [comment]
Sure +1

**Comment by earthmoonjune** [comment]
+1

**Comment by Hate9** [comment]
@t-f-w Wait, did you just +1 a merged PR?

**Comment by Carbonhell** [comment]
Amazing

**Comment by Carbonhell** [comment]
He clearly loves the feature enough he feels the need to express it with his +1... Be thankful smh.......

**Pull request merged**